python/tank/descriptor/io_descriptor/base.py

```diff
@@ -40,12 +40,47 @@ class IODescriptorBase(object):
     Tank App store and one which knows how to handle the local file system.
     """
 
-    def __init__(self, descriptor_dict):
+    _factory = {}
+
+    @classmethod
+    def register_descriptor_factory(cls, descriptor_type, subclass):
+        """
+        Registers a descriptor subclass with the :meth:`create` factory.
+        This is an internal method that should not be called by external
```

Comment:
`IODescriptor` is already an internal class, so no need to mention that it is an internal method.